### PR TITLE
Improve backend timestamp parsing

### PR DESCRIPTION
**Pull request by carlanton**
possible fix for #350

### CODE REVIEW

File: pkg/parser.go [hunk 1]
```diff
@@ -273,6 +273,9 @@ func ParseValue(fieldName string, fieldType string, tz *time.Location, value int
 
 			return parseUInt64Value(value, isNullable)
 		case "Int64":
+			if fieldName == "t" {
```
Comment (Slach) [inline pkg/parser.go:276]: could you explain? do you use toInt64(dateTimeField) in your SQL query?

Current macros for DateTime handling usually use toUInt32

File: pkg/response.go [hunk 2]
```diff
@@ -78,9 +78,11 @@ func (r *Response) toFramesWithTimeStamp(query *Query, fetchTZ FetchTZFunc, hasL
 	valueDataFieldMap := map[string]*data.Field{}
 
 	for _, row := range r.Data {
-		timestampValue := ParseValue(
-			timestampFieldName, timestampFieldType, timeZonesMap[timestampFieldName], row[timestampFieldName], false,
-		).(time.Time)
+		value := ParseValue(timestampFieldName, timestampFieldType, timeZonesMap[timestampFieldName], row[timestampFieldName], false)
+		timestampValue, ok := value.(time.Time)
+		if !ok {
```
Comment (Slach) [inline pkg/response.go:83]: thanks nice catch

File: pkg/parser.go (hunk 1 above)
Comment (carlanton) [reply pkg/parser.go:276]: The query is generated by the backend! I don't think I'm doing anything special? Could it be related to that I'm using datetime64?
Check out https://github.com/Vertamedia/clickhouse-grafana/issues/350 for more details.

Comment (bmanth60) [reply pkg/parser.go:276]: Yeah, the docs explicitly state DateTime64 is stored as Int64 https://clickhouse.tech/docs/en/sql-reference/data-types/datetime64/amp/

**Pull request merged**